pyrovision/datasets/openfire.py

```diff
@@ -4,195 +4,69 @@
-        """
-
-        # Check extract existence
-        file_path = self._annotations.joinpath(self.url.rpartition('/')[-1])
-        if not file_path.is_file():
-            raise RuntimeError('Extract not found. You can use download=True to download it.')
-        # Take the specified number of samples
-        with open(file_path, 'rb') as f:
-            extract = json.load(f)[:num_samples]
-
-        return extract
-
-    def _download_images(self, extract, threads=None):
-        """ Download images from a specific extract
-
-        Args:
-            extract (list<dict>): image extract to download
-            threads (int, optional): number of threads used for parallel downloading
-        """
-
-        self._images.mkdir(parents=True, exist_ok=True)
-        # Prepare URL and filenames for multi-processing
-        entries = [(s['url'], s['name']) for s in extract
-                   if not self._images.joinpath(s['name']).is_file()]
-        # Use multiple threads to speed up download
-        if len(entries) > 0:
-            download_urls(entries, self._images, threads=threads)
-
-    def _verify_samples(self, extract):
-        """ Download images from a specific extract
-
-        Args:
-            extract (list<dict>): list of samples
-        Returns:
-            valid_samples (list<dict>): list of valid samples
-        """
-
-        valid_samples = []
-        dl_issues, target_issues = 0, 0
-        # Verify samples in extract
-        for sample in extract:
-
-            is_ok = True
-            # Verify image
-            if not self._images.joinpath(sample['name']).is_file():
-                dl_issues += 1
-                is_ok = False
-
-            # Verify targets
-            if self.class_to_idx.get(sample['target']) is None:
-                target_issues += 1
-                is_ok = False
-
-            if is_ok:
-                valid_samples.append(sample)
-
-        # HTTP errors
-        if dl_issues == len(extract):
-            raise RuntimeError('Images not found. You can use download=True to download them.')
-        elif dl_issues > 0:
-            warnings.warn(f'{dl_issues}/{len(extract)} sample images are not present on disk. '
-                          'Please retry downloading later.')
-        # Extract errors
-        if target_issues > 0:
-            warnings.warn(f'{target_issues}/{len(extract)} samples have corrupted targets.')
-
-        return valid_samples
+        if self.train:
+            self.path_to_imgs = Path(self.root, 'train')
+        else:
+            self.path_to_imgs = Path(self.root, 'test')
 
-    def extra_repr(self):
-        return "Split: {}".format("Train" if self.train is True else "Test")
+        super(OpenFire, self).__init__(self.path_to_imgs, loader, None,
+                                       transform=transform,
+                                       target_transform=target_transform,
+                                       is_valid_file=loader)
```

Comment:
I'm not really familiar with DatasetFolder. Do you have any ref of implemented dataset of torchvision using it so I can dive into this better please ? :pray:

Reply:
I guess you know [ImageFolder](https://github.com/pytorch/vision/blob/4ec38d496db69833eb0a6f144ebbd6f751cd3912/torchvision/datasets/folder.py#L177), which is [recommended by god](https://discuss.pytorch.org/t/custom-label-for-torchvision-imagefolder-class/52300/2) by the way :) The goal is to adapt the ImageFolder construction to our problem.